src/CreateX.sol

```diff
@@ -911,8 +905,9 @@ contract CreateX {
             // Reverts if the 21st byte is greater than `0x01` in order to enforce developer explicitness.
             revert InvalidSalt({emitter: _SELF});
         } else {
-            // In all other cases, the salt value `salt` is not modified.
-            guardedSalt = salt;
+            // For the non-pseudo-random cases, the salt value `salt` is hashed to prevent the safeguard mechanisms
+            // from being bypassed. Otherwise, the salt value `salt` is not modified.
+            guardedSalt = (salt != _generateSalt()) ? keccak256(abi.encode(salt)) : salt;
```

Comment:
Seems we could just always hash the salt to simplify. For the generated salt case it gets double hashed which is ok, and it's cheaper to just hash that then recompute `_genrateSalt()`. Is there a downside to this?
```suggestion
            // Hash the salt value to prevent safeguard mechanisms from being bypassed.
            guardedSalt = keccak256(abi.encode(salt));
```

Reply:
If you hash the generated salt then someone could supply the same generated salt on another chain manually, it will get hashed and result in the same final salt.

Reply:
Exactly, this would open up the attack vector again.

Reply:
Ah yes that’s correct